testdata/project-v4-multigroup/internal/controller/certmanager/certificate_controller.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2024 The Kubernetes authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certmanager
+
+import (
+	"context"
+
+	certmanagerv1 "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// CertificateReconciler reconciles a Certificate object
+type CertificateReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=certmanager.testproject.org,resources=certificates,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=certmanager.testproject.org,resources=certificates/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=certmanager.testproject.org,resources=certificates/finalizers,verbs=update
```

Comment:
@fischor 


Thank you for your comment https://github.com/kubernetes-sigs/kubebuilder/pull/4171#issuecomment-2367159862
I will give a look over the rbac marker for the controller-gen that is a good point.

PS.: It would be helpful you create comments based on the code and not like you are doing since it is impossible follow a tread

Reply:
Done 👍 
Thank you for the catcher 🎉